bakalaurinis.tex

```diff
@@ -0,0 +1,99 @@
+\university{Vilniaus universitetas}
+\faculty{Matematikos ir informatikos fakultetas}
+% \institute{Informatikos institutas}  % Užkomentavus šią eilutę - institutas neįtraukiamas į titulinį
+\department{Programų sistemų katedra}
+\title{Daugiaplatformių mobiliųjų aplikacijų kūrimo karkasų analizė}
+\titleineng{Analysis of cross-platform mobile app development frameworks}
+\papertype{Bakalaurinio darbo planas}
+\status{4 kurso 4 grupės studentas}
+\author{Edas Lakavičius}
+% \secondauthor{Vardonis Pavardonis}   % Pridėti antrą autorių
+\supervisor{lekt. Jonas Ragaišis}
+\date{Vilnius \\ \the\year}
+
+% Nustatymai
+% \setmainfont{Palemonas}   % Pakeisti teksto šriftą į Palemonas (turi būti įdiegtas sistemoje)
+\bibliography{bibliografija} 
+
+\begin{document}
+\maketitle
+
+\tableofcontents
+
+\section{Tyrimo objektas ir aktualumas}
+Tyrimo objektas - daugiaplatformių mobiliųjų aplikacijų kūrimo karkasai.
+
+Šiais laikais internetas, socialinė medija ir elektroninė prekyba yra neatsiejami nuo kasdienybės dalykai. Interneto naudojimas išaugo nuo 413 klientų milijonų 2000 metais iki 3.4 bilijonų 2016 \cite{owidinternet} ir vis dar auga beprotišku greičiu kiekvieną dieną. Didžioji dalis interneto naudotojų yra per mobilius telefonus prisijungiantys klientai, kurie greit lenks kompiuterius naudojančius klientus, net 4 kartus \cite{internetusage}. Būtent dėl to nenuostabu, kad atsiranda vis daugiau projektų, kurie fokusuojasi į mobiliųjų aplikacijų kūrimą.
+
+Šiuo metu didžiausios aplikacijų parduotuvės "Google Play" ir "Apple App Store" turi daugiau nei 5 milijonų mobiliųjų aplikacijų \cite{appcount} ir įmonei, norinčiai efektyviai dalyvauti šioje rinkoje reiks ne tik daug investuoti, bet ir gebėti greitai kurti ir gerinti savo aplikacijas. Įprastai yra kuriamos atskiros aplikacijos "iOS" ir "Android" platformom, kas dažniausiai reikalauja turėti atskiras komandas ar programuotojus specializuojančius prie vienos iš šių sistemų, tačiau pačios aplikacijos turi būti kuriamos vienodos.
+
+Būtent dėl didelių kaštų, aplikacijų kūrimo greičio ir skirtingų platformų apjungimo išpopuliarėjo hibridinio tipo technologijos - daugiaplatformių mobiliųjų aplikacijų kūrimo karkasai. Per trumpą laiką buvo sukurta gana daug skirtingų technologijų skirtų tam pačiam prieš tai jau minėtam tikslui - apjungti skirtingas platformas. Todėl yra aktualu šiuos karkasus išanalizuoti ir palyginti, kad skirtingi projektai galėtų pasirinkti sau tinkamiausią.
+
+
+\section{Darbo tikslas}
+Darbu siekiama, atliekant skirtingų daugiaplatformių aplikacijų kūrimo karkasų analizę, sukurti rekomendacijų rinkinį, padėsiantį pasirinkti tarp išanalizuotų variantų.
+
+\begin {section}{Keliami uždaviniai ir laukiami rezultatai}
+Keliami uždaviniai:
+\begin{itemize}
+  \item Sukurti bazines aplikacijas, naudojant pasirinktus karkasus.
+  \item Aprašyti pasirinktų aplikacijų kūrimo karkasų architektūras.
+  \item Išanalizuoti pasirinktų aplikacijų kūrimo karkasų privalumus bei trūkumus.
```

Comment:
Privalumai ir trūkumai nėra tinkamas rezultatas / tikslas bakalauro darbui, o tai paskutinis uždavinys, kuris turėtų būti arčiausiai tikslo pasiekimo. Manau, kad reikia dar pagalvoti dėl tikslo ir uždavinių, kad jie tenkintų mokslo tiriamojo darbo reikalavimus. 
Čia nuoroda kur trumpai aprašo kas yra mokslo tiriamasis darbas 
https://perevod.lt/blog/2016/01/17/mokslinis-tiriamasis-darbas/

Čia darbas, kuris vertina eksploatacines savybes https://epublications.vu.lt/object/elaba:81705193/MAIN
Tai ganėtinai konkretu, nurodoma problematika, tikslas, uždaviniai ir panašiai. Galima remtis kaip pavyzdžiu formuluojant uždavinius

Reply:
Kaip suprantu reiktų konkretesnius turėti kriterijus ties kuriais lyginsiu? Pvz jūsų pateiktame pavyzdiniame darbe atsiremia į performance.

Reply:
Aš šiek tiek plačiau įsivaizdavau būtų galima palyginti, bet nebuvau apgalvojęs konkrečių kriterijų. Galima būtų remtis šita lentele:


Parameter | Native Apps | Cross-Platform Apps
-- | -- | --
Cost | High cost of development | Relatively low cost of development
Code Usability | Works for a single platform | Single code can be used on multiple platform, for an easy portability
Device Access | Platform SDK ensures access to device’s API without any hindrance | No assured access to all device APIs
UI Consistency | Consistent with the UI components of the device | Limited consistency with the UI components of the device
Performance | Seamless performance, given the app is developed for the devices OS | High on performance, but lags and hardware compatibility issues are not uncommon

Reply:
remiantis šia lentele vis tiek gana abstrakčiai gaunasi. Kaip kainą įvertinti? Ką reiškia High cost arba relatively low? O maintenance kaina? ji irgi labai svarbi bendrai projekto išlaidoms. Principe pabaigoje galima kažką tokio sukurti kaip apibendrinimą, tačiau jei vien tik tokia lentelė, tai ji nelabai kuo skiriasi nuo kitų tinklaraščiuose randamų palyginimų. Kiekvienas iš šių lentelės aspektų gali būti gana plačiai aprašinėjamas. Pvz. į kainą įeina ir kiek populiarus framework, vidutinė metinė alga tos srities specialisto, lengvumas surasti tuos specialistus, greitis kaip greitai hot reload padaro, kiek problemų būna su native integracijom (pvz. ar kamera tvarkingai veikia ant bet kokio įrenginio ar yra problemų) ir t.t. ir panašiai. Tame darbe fokusuojasi į performance ir moksliškai bando pagrįsti - tai jau geras kelias. Todėl svarbu, kad nebūtų per daug abstraktus ir platus tikslas. Aišku jį galima konkretizuoti, tačiau nuo tikslo ir uždavinių šiek tiek priklauso ir įsivaizdavimas apie ką rašysi, todėl nėra blogai ir iš anksto apie tai pagalvoti

Reply:
Hmm, bet aš kažkaip ir norėjau imti abstrakčiau, kad būtų galima palyginti daugmaž pagrindinius aspektus - performance, development speed, architecture ir limitations. Tas cost tai gerai pastebėta, tikrai labai neapibrėžtas. Nežinau dabar truputį pasimečiau kuria linkme čia būtų galima eiti.

Reply:
Tai dabar atrodo pagrindinis blockeris, kad reikia sugalvoti pagal ką aš lyginsiu frameworkus ir darysiu tą rekomendacijų rinkinį, iš to atitinkamai ir bus galima konkretizuoti užduotis, tikslą ir rezultatus.

Reply:
Kadangi pagrindinė idėja ir buvo sukurti tokį rekomendacijų rinkinį pagal kurį projekto pradžioje būtų galima atitinkamai pasirinkti tinkamiausią cross-platform framework'ą, tai atrodo ir reikia plačiai analizuoti.